### PR TITLE
Feature: atmosphere shader

### PR DESCRIPTION
**Pull request by smlpt**
Implements a dynamic atmosphere shader, based on the Nishita sky model. The code is ported from https://github.com/wwwtyro/glsl-atmosphere. 
An example can be found in basic/AtmosphereExample.

- [x] (dynamically hand over sun position data via @ShaderProperty)
- [x] Implement key binds for sun position (via dragsphere behavior)
- [x] ~~write vertex shader for sunproxy sphere to disable translational movement~~ circumvented dragbehavior with keybinds
- [x] ~~Synchronize a directional light to the sun position~~
- [x] Create a default atmosphere object that can be imported to automatically set up an atmosphere
- [x] Add an emission material property to the render pipeline so that the atmosphere object does not require to be lit by point lights
- [x] Create vertex shader to to always put the icosphere in the background like a skybox (see learnopengl.com)
- [x] local time to sun position
- [x] update sun position with coroutine
- [x] clean up code

### CODE REVIEW

File: src/main/kotlin/graphics/scenery/PointLight.kt [hunk 1]
```diff
@@ -28,6 +28,12 @@ class PointLight(val radius: Float = 5.0f) : Light("PointLight") {
     @ShaderProperty
     override val lightType: LightType = LightType.PointLight
 
+    /** Introduces self-lighting emission for the parent objects material.
+     * This attribute should only be used when the [PointLight] acts as a proxy for the corresponding @ShaderProperty. */
+    //@ShaderProperty
+    //var emissive: Float = 1f
+    //TODO Remove this part
```
Comment (skalarproduktraum) [inline src/main/kotlin/graphics/scenery/PointLight.kt:35]: Follow this TODO, you must 😁

File: src/main/resources/graphics/scenery/backends/shaders/DeferredLighting.frag [hunk 2]
```diff
@@ -60,6 +61,8 @@ layout(set = 5, binding = 0, std140) uniform ShaderProperties {
     vec3 worldPosition;
     vec3 emissionColor;
     int lightType;
+    // self-lighting property of a material
```
Comment (skalarproduktraum) [inline src/main/resources/graphics/scenery/backends/shaders/DeferredLighting.frag:64]: Remove

**Review by skalarproduktraum — CHANGES_REQUESTED** [review]
Thanks, only some minor stuff left 👍

File: src/main/resources/graphics/scenery/backends/shaders/DeferredLighting.frag [hunk 3]
```diff
@@ -485,6 +488,7 @@ void main()
 	vec3 N = DecodeOctaH(texture(InputNormalsMaterial, textureCoord).rg);
 
 	vec4 Albedo = texture(InputDiffuseAlbedo, textureCoord).rgba;
+    vec4 Emissive = texture(InputEmission, textureCoord).rgba;
 	float Specular = texture(InputDiffuseAlbedo, textureCoord).a;
```
[the remark this answers is not in the document]
Comment (smlpt) [reply src/main/resources/graphics/scenery/backends/shaders/DeferredLighting.frag:492]: its consistent in IntelliJ so I'll keep it as is.

File: src/main/resources/graphics/scenery/backends/shaders/sunProxy.vert [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+} currentEye;
+
+void main()
+{
+    mat4 viewMatrix;
+    mat4 nMVP;
+    mat4 projectionMatrix;
+
+    viewMatrix = (vrParameters.stereoEnabled ^ 1) * ViewMatrices[0] + (vrParameters.stereoEnabled * ViewMatrices[currentEye.eye]);
+    //viewMatrix[3] = vec4(0.0f, 0.0f, 0.0f, 1.0f);
```
[the remark this answers is not in the document]
Comment (smlpt) [reply src/main/resources/graphics/scenery/backends/shaders/sunProxy.vert:48]: I will completely remove sunproxy.vert, as it isn't used at all anymore.

**Review by skalarproduktraum — APPROVED** [review]

**Review by skalarproduktraum — APPROVED** [review]
Thanks, @smlpt 👍

**Pull request merged**